File: monkeyKit/src/test/kotlin/com/criptext/security/ShadowAESUtil.java
package com.criptext.security;

import android.content.Context;
import org.robolectric.annotation.Implementation;
import org.robolectric.annotation.Implements;

/**
 * Created by gesuwall on 12/23/16.
 */

@Implements (AESUtil.class)
public class ShadowAESUtil {

        @Implementation
        public static String decryptWithCustomKeyAndIV(String encryptedText, String key, String iv) {
            return "decrypted:" + encryptedText;

        }

    @Implementation
    public void __constructor__(String keyAndIV) {

    }


    @Implementation
    public void __constructor__(Context ctx, String keyAndIV) {

    }

    @Implementation
    public String decrypt(String inputString){
        return inputString;
    }

}

File: monkeyKit/src/test/kotlin/com/criptext/security/ShadowAESUtil.kt
package com.criptext.security

import android.content.Context
import org.robolectric.annotation.Implementation
import org.robolectric.annotation.Implements

/**
 * Created by gesuwall on 12/23/16.
 */

@Implements (AESUtil::class)
class ShadowAESUtil {

    companion object {
        @Implementation
        fun decryptWithCustomKeyAndIV(encryptedText: String, key: String, iv: String) {

        }
    }

    @Implementation
    fun __constructor__(keyAndIV: String) {

    }


    @Implementation
    fun __constructor__(ctx: Context?, keyAndIV: String) {

    }

    @Implementation
    fun decrypt(inputString: String): String {
        return inputString
    }

}